src/runtime/pprof/pprof.go

```diff
@@ -776,6 +779,107 @@ func writeGoroutineStacks(w io.Writer) error {
 	return err
 }
 
+// writeGoroutineStacksWithLabels writes individual goroutine stack traces like
+// writeGoroutineStacks, but uses the concurrent, reduced-stop time approach of
+// goroutineProfileWithLabelsConcurrent. It includes the ID, state, and labels
+// for each goroutine, although as these are captured after the world is resumed
+// they are not guaranteed to be entirely consistent as of a single point in
+// time.
+func writeGoroutineStacksWithLabels(w io.Writer) error {
+	n, ok := pprof_goroutineStacksWithLabels(nil, nil, nil)
+
+	for {
```

Comment:
are you confident in the robustness of this loop? Maybe worth to additionally put in a growth factor of (say) 2 on each iteration.